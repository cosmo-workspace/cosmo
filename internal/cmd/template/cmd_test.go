package template

import (
	"bytes"
	"testing"

	"github.com/cosmo-workspace/cosmo/pkg/cli"
	. "github.com/cosmo-workspace/cosmo/pkg/snap"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"

	"github.com/spf13/cobra"
)

func TestCommandTemplate(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "cosmoctl template suite")
}

var _ = Describe("help", func() {
	Context("template", func() {
		It("should match snapshot", func() {
			cmd := &cobra.Command{}
			out := bytes.Buffer{}
			cmd.SetOut(&out)
			AddCommand(cmd, cli.NewRootOptions())
			cmd.SetArgs([]string{"template", "--help"})
			err := cmd.Execute()
			Expect(err).ShouldNot(HaveOccurred())
			Expect(out.String()).To(MatchSnapShot())
		})
	})

	Context("template generate", func() {
		It("should match snapshot", func() {
			cmd := &cobra.Command{}
			out := bytes.Buffer{}
			cmd.SetOut(&out)
			AddCommand(cmd, cli.NewRootOptions())
			cmd.SetArgs([]string{"template", "generate", "--help"})
			err := cmd.Execute()
			Expect(err).ShouldNot(HaveOccurred())
			Expect(out.String()).To(MatchSnapShot())
		})
	})

	Context("template get", func() {
		It("should match snapshot", func() {
			cmd := &cobra.Command{}
			out := bytes.Buffer{}
			cmd.SetOut(&out)
			AddCommand(cmd, cli.NewRootOptions())
			cmd.SetArgs([]string{"template", "get", "--help"})
			err := cmd.Execute()
			Expect(err).ShouldNot(HaveOccurred())
			Expect(out.String()).To(MatchSnapShot())
		})
	})
})
