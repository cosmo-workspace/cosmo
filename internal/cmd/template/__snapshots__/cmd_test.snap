['help should match snapshot 1']
SnapShot = """

Manipulate COSMO Workspace Template resource.

\"Template\" is a set of Kubernetes resources for Workspace.

Usage:
   template [command]

Aliases:
  template, tmpl

Available Commands:
  generate    Generate Template
  get         Get Templates
  validate    Validate Template by dry-run

Flags:
  -h, --help   help for template

Use \" template [command] --help\" for more information about a command.
"""

['help template generate should match snapshot 1']
SnapShot = """
Generate Template

Usage:
   template generate [command]

Aliases:
  generate, gen

Available Commands:
  useraddon   Generate UserAddon
  workspace   Generate WorkspaceTemplate

Flags:
  -h, --help   help for generate

Use \" template generate [command] --help\" for more information about a command.
"""

['help template get should match snapshot 1']
SnapShot = """
Get Templates

Usage:
   template get [command]

Aliases:
  get, list

Available Commands:
  useraddons  Get addons
  workspace   Get workspace templates in cluster

Flags:
  -h, --help   help for get

Use \" template get [command] --help\" for more information about a command.
"""

['help template should match snapshot 1']
SnapShot = """

Manipulate COSMO Workspace Template resource.

\"Template\" is a set of Kubernetes resources for Workspace.

Usage:
   template [command]

Aliases:
  template, tmpl

Available Commands:
  generate    Generate Template
  get         Get Templates
  validate    Validate Template by dry-run

Flags:
  -h, --help   help for template

Use \" template [command] --help\" for more information about a command.
"""
